statescript/executor.go

```diff
@@ -206,14 +327,40 @@ func (l Launcher) ExecuteAll(state, action string, ignoreError bool) error {
 			}
 		}
 
-		if ret := execute(filepath.Join(dir, s.Name()), timeout); ret != 0 {
-			// In case of error scripts all should be executed.
-			if ignoreError {
-				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
-			} else {
-				return errors.Errorf("statescript: error executing '%s': %d",
-					s.Name(), ret)
+		retryLaterFlag := false
+
+		for { // retry-later
```

Comment:
I think I would simplify this a bit here. You can leave the `for` loop, but how about something like:
```
for {
      switch ret := execute(filepath.Join(dir, s.Name()), timeout) {
      case 0:
        // success; just break the loop and continue
        break
      case exitRetryLater:
        // we need to re-run the script later
        time.Sleep(l.getRetryInterval())
        continue
      default:
        // some error occured
        
        // In case of error scripts all should be executed.
  			if ignoreError {
  				log.Errorf("statescript: ignoring error executing '%s': %d", s.Name(), ret)
          break
  			} else {
  				return errors.Errorf("statescript: error executing '%s': %d",
  					s.Name(), ret)
  			}
      }
    }
```
I think the above might cover the whole functionality we need ATM. I am not sure why we need to store some context inside the DB and read this all the time. All we need is probably busy waiting here.